contracts/test/Revertal.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * This file ensures that the provider returns the same error message structure as Geth for kinds of revert errors.
+ *
+ * | revert fashion in Solidity | evm.result.ReturnData | jsonrpc.result |
+ * | :--- | :--- | :--- |
+ * | revert() | "" | { code: -32000, message: 'execution reverted' } |
+ * | revert("reason") | RevertErrorSelector + DataOffset + StringLength + StringData | { code: -32000, message: 'execution reverted: reason', data: evm.result.ReturnData.hexadecimal()" } |
+ * | revert CustomError("reason") | ... | { code: -32000, message: 'execution reverted' } |
+ *
+ * Ref:
+ * - https://docs.soliditylang.org/en/latest/contracts.html#errors
+ * - https://docs.soliditylang.org/en/latest/control-structures.html#error-handling-assert-require-revert-and-exceptions
+ * - https://www.tutorialspoint.com/solidity/solidity_error_handling.htm
+ *
+ * Note:
+ * - Discard the `code` field in the JSONRPC response, as Godwoken's error codes are different from Geth
+ * - Discard the `code` field in `error` in the JSONRPC response, as Godwoken's error codes are different from Geth
+ */
+
+const {expect} = require("chai");
+const {network, ethers} = require("hardhat");
+const {isGwMainnetV1} = require('../utils/network');
+const {fetchJson} = require("ethers/lib/utils");
+
+describe("Revertal", function () {
+    // Skip for gw_mainnet_v1 network
+    if (isGwMainnetV1()) {
+        return;
+    }
+    // Skip for hardhat network
+    if (network.name === "hardhat") {
```

Comment:
Can this be added to "../utils/network" for more convenient usage search?

Reply:
Sounds good. Like the below code?

```ts
function isHardhatNetwork() {
  return isNetwork(Networks.Hardhat);
}
```

Reply:
> Sounds good. Like the below code?
> 
> ```
> function isHardhatNetwork() {
>   return isNetwork(Networks.Hardhat);
> }
> ```

Yes, it should be easier for us to search for where it is used.